File: Tools/Annotation/Labeling/api_handling/api_wrapper/projects.py
import requests
import traceback

from api_wrapper.main import base_url, auth

def get_all_projects():
    try:
        response = requests.get(f"{base_url()}/get_projects", auth = auth())  # Adjust the endpoint as needed
        return response.json()
    except Exception:
        return {"error": True, "value": "Failed to make request", "stack": traceback.format_exc()}

def get_project(pid):
    try:
        response = requests.get(f"{base_url()}/get_project/{pid}", auth = auth())  # Adjust the endpoint as needed
        return response.json()
    except Exception:
        return {"error": True, "value": "Failed to make request", "stack": traceback.format_exc()}

def create_project(project_name):
    try:
        response = requests.post(f"{base_url()}/create_project", json={"project_name": project_name}, auth = auth())
        return response.json()
    except Exception:
        return {"error": True, "value": "Failed to make request", "stack": traceback.format_exc()}

def delete_project(project_id):
    try:
        response = requests.post(f"{base_url()}/delete_project", json={"project_id": project_id}, auth = auth())
        return response.json()
    except Exception:
        return {"error": True, "value": "Failed to make request", "stack": traceback.format_exc()}

def rename_project(project_id, new_name):
    try:
        response = requests.post(f"{base_url()}/rename_project", json={"project_id": project_id, "project_name": new_name}, auth = auth())
        return response.json()
    except Exception:
        return {"error": True, "value": "Failed to make request", "stack": traceback.format_exc()}

File: Sewing/Tools/Annotation/Labeling/set_up_routes/data_routes/backend/projects.js
const project_data_actions = require("../data_actions/projects.js");

module.exports = (app) => {
    app.post(`/create_project`, async (req, res) => {
        try {
            const r = await project_data_actions.create_project(req.body.project_name);
            return res.json({
                error: false,
                ...r
            });
        } catch (err){
            return res.json({
                error: true,
                value: err.toString(),
                trace: err.stack
            })
        }
    });

    app.post(`/rename_project`, async (req, res) => {
        try {
            const r = await project_data_actions.rename_project(req.body.project_id, req.body.project_name);

            return res.json({
                error: false,
                ...r
            });
        } catch (err){
            return res.json({
                error: true,
                value: err.toString(),
                trace: err.stack
            })
        }
    });

    app.post(`/delete_project`, async (req, res) => {
        try {
            const r = await project_data_actions.delete_project(req.body.project_id);
            return res.json({
                error: false
            });
        } catch (err){
            return res.json({
                error: true,
                value: err.toString(),
                trace: err.stack
            });
        }
    });

    app.get(`/get_project/:project_id`, async (req, res) => {
        try {
            const r = await project_data_actions.get_project(req.params.project_id);
            return res.json({
                error: false,
                ...r
            });
        } catch (err){
            return res.json({
                error: true,
                value: err.toString(),
                trace: err.stack
            });
        }
    });

    app.get(`/get_projects`, async (req, res) => {
        try {
            const r = await project_data_actions.get_projects();
            return res.json({
                error: false,
                projects: r
            });
        } catch (err){
            return res.json({
                error: true,
                value: err.toString(),
                trace: err.stack
            });
        }
    });
}